wordpress-mobile/WordPress-iOS
# My Site Dashboard: fix breaking constraints

Part of #17873 

This PR fixes breaking constraints under the Dashboard.

Basically, what was happening is that the `UICollectionView` layout calculation adds a height to each and every cell when performing calculations.

However, some of the views we appended had their own intrinsic height (like `UIStackView`). The system was then unable to determine which height to follow, given the cell was telling one and the view inside the cell was telling other values.

### Additional info

I had to add a "minimum height" on the posts view controller to avoid this:

https://user-images.githubusercontent.com/7040243/157529587-ba57ce42-2199-48e2-90c3-e77b107884d7.mp4

### To test

1. Run the app with MSD flag enabled
2. Tap "Home"
3. ✅ Check the console output, it shouldn't have any constraint warnings (enjoy the quietness)
4. ✅ The dashboard and it's cells should appear correctly (and more smoothy 🧈) 

### Out of scope

We still have the `[TableView] Warning once only: UITableView was told to layout its visible cells and other contents without being in the view hierarchy` warnings being triggered, however this one is out of scope for this task.

## Regression Notes
1. Potential unintended areas of impact
n/a

2. What I did to test those areas of impact (or what existing automated tests I relied on)
n/a

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:
